### PR TITLE
Allow for multiple host names (#84)

### PR DESCRIPTION
**Pull request by marians**
Based on https://github.com/giantswarm/oauth2-proxy-app/pull/84 by @johannges

## How this should be tested

- New workload cluster with NGINX ingress controller and cert-manager installed
- Two host names configured
- Auth0 as OAuth provider

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,22 +45,25 @@ More options can be found in the [command line options documentation](https://oa
...
-        - path: /oauth2
-            backend:
-            serviceName: oauth2-proxy
-            servicePort: 4180
+          paths:
+          - backend:
+              service:
+                  name: oauth2-proxy
+                  port:
+                  number: 4180
```
Comment (ubergesundheit) [inline README.md:53]:
```suggestion
                    number: 4180
```

File: README.md [hunk 2]
```diff
@@ -45,22 +45,24 @@ More options can be found in the [command line options documentation](https://oa
...
-            backend:
-            serviceName: oauth2-proxy
-            servicePort: 4180
+          paths:
+          - backend:
+              service:
+                name: oauth2-proxy
+                port:
+                  number: 4180
+            path: /oauth2
```
Comment (ubergesundheit) [inline README.md:54]:
```suggestion
            path: /oauth2
            pathType: Prefix
```

**Review by ubergesundheit — APPROVED** [review]
LGTM once values schema has been updated https://intranet.giantswarm.io/docs/organizational-structure/teams/cabbage/app-updates/helm-values-schema/

**Comment by anvddriesch** [comment]
Tested using auth0 and cluster on gauss with nginx-ingress-controller, cert-manager and two different hello-world apps with different hosts.

**Pull request merged**